src/aggregation/self_attention_pool.py

```diff
@@ -104,8 +104,11 @@ def forward(self, x, attn_masks=None):
         for attn, ff in self.layers:
             x = attn(x, attn_mask=attn_masks) + x
             x = ff(x) + x
-
-        x = torch.mean(x, 1)
+        if attn_masks is not None: # masked mean
+            denom = torch.sum(attn_masks, dim=-1, keepdim=True)
```

Comment:
do we need this? is this tested?

Reply:
it worked to take the spikes out of the training loss curve and i tested the masked mean part locally, but i can do some further testing on this